### PR TITLE
rtx global shell local

### PR DESCRIPTION
**Pull request by dfang**
#711

### CODE REVIEW

**Comment by dfang** [comment]
@jdxcode  for 3, I never used `nushell` and `xnosh`, can you help?
and also 4 is a little tough hard for me.

File: src/shell/bash.rs [hunk 1]
```diff
@@ -29,7 +29,11 @@ impl Shell for Bash {
 
               case "$command" in
               deactivate|shell)
-                eval "$(command rtx "$command" "$@")"
+                if [ "$1" = "-h" ]; then
```
Comment (jdx) [inline src/shell/bash.rs]: This is missing some cases like:
* `rtx shell --help`
* `rtx shell node@18 -h`

Comment (dfang) [reply src/shell/bash.rs]: cases:
- `rtx shell -h`
- `rtx shell --help`
- `rtx shell node@18 -h`


bash and zsh

```
STR1="rtx shell --help"
STR2="rtx shell node@18 -h"
STR3="rtx shell -h"
SUB='-h'
if [[ "$STR1" == *"$SUB"* ]]; then
  echo "yes"
fi
if [[ "$STR2" == *"$SUB"* ]]; then
  echo "yes"
fi
if [[ "$STR3" == *"$SUB"* ]]; then
  echo "yes"
fi
```

fish: 
```
set STR1 "rtx shell --help"
set STR2 "rtx shell node@18 -h"
set STR3 "rtx shell -h"
set substring_to_find "*-h*"
if string match -q $substring_to_find $STR1
    echo "yes"
end
if string match -q $substring_to_find $STR2
    echo "yes"
end
if string match -q $substring_to_find $STR3
    echo "yes"
end
```

hi @jdxcode, I've added some sample script here, PTAL.

**Comment by notpeelz** [comment]
This fails with `rtx shell -vhy` and other niche scenarios.

Try this instead (zsh/bash):
```diff
   case "$command" in
   deactivate|shell)
+    for arg in "$@"; do
+      if [[ "$arg" == "--help" || "$arg" =~ ^-[a-zA-Z]*h.*$ ]]; then
+        command rtx "$command" "$@"
+        return
+      fi
+    done
     eval "$(command rtx "$command" "$@")"
     ;;
```

I've never used fish so I don't know what the equivalent syntax would be.

**Comment by jdx** [comment]
superseded by https://github.com/jdx/rtx/pull/976

**Pull request closed without merging**